overlord/servicestate/quota_handlers.go

```diff
@@ -416,8 +416,8 @@ func quotaRemove(st *state.State, action QuotaControlAction, allGrps map[string]
 		return nil, nil, false, fmt.Errorf("internal error, AddSnaps option cannot be used with remove action")
 	}
 
-	if action.ResourceLimits.Memory != nil {
-		return nil, nil, false, fmt.Errorf("internal error, MemoryLimit option cannot be used with remove action")
+	if !action.ResourceLimits.Unset() {
+		return nil, nil, false, fmt.Errorf("internal error, quota limit options cannot be used with remove action")
```

Comment:
nitpick: we tend to write "internal error: ..." with  a colon and not a comma. If we change it we should change the other ones here as well but as I said nitpick